### PR TITLE
Separate search and shoebox modules and their dependencies

### PR DESCRIPTION
**Pull request by gmethvin**
Just trying to get some feedback on my organization. **Do not merge**, since this hasn't been fully tested.

The idea is to have a separate Guice module for search so we can create a separate search server using the search module. I also separated out some of the other dependencies and some repeated code between dev and prod mode.

### CODE REVIEW

File: shoebox/app/com/keepit/dev/DevGlobal.scala [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
+import com.keepit.shoebox.ShoeboxModule
+import play.api.Application
+import play.api.Mode._
+import play.api.Play.current
 
 object DevGlobal extends FortyTwoGlobal(Dev) {
 
-  override lazy val injector: Injector = Guice.createInjector(Stage.DEVELOPMENT, new DevModule())
+  override lazy val injector: Injector = Guice.createInjector(Stage.DEVELOPMENT,
+  Modules.`override`(new ShoeboxModule(Dev)).`with`(new SearchModule(Dev)))
```
Comment (2is10) [inline shoebox/app/com/keepit/dev/DevGlobal.scala]: I really like the direction you're going, breaking up modules for reuse and avoiding duplication. One thing I'd really recommend avoiding, though, is parameterizing modules with `Mode`. Slippery slope. Instead, how about we write modules for production and then define and apply dev and test modules as overrides. The `Modules.override` [javadoc](http://google-guice.googlecode.com/svn/trunk/javadoc/com/google/inject/util/Modules.html#override%28com.google.inject.Module...%29) illustrates exactly this pattern:

```
Module functionalTestModule
     = Modules.override(new ProductionModule()).with(new TestModule());
```

Comment (gmethvin) [inline shoebox/app/com/keepit/dev/DevGlobal.scala]: Yeah, I'll change this. I think this is the kind of thing overrides should be used for.

File: shoebox/app/com/keepit/module/S3Module.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import com.keepit.common.logging.Logging
+import com.keepit.common.social.S3SocialUserRawInfoStoreImpl
+import com.keepit.common.social.SocialUserRawInfoStore
+import com.keepit.common.store.S3Bucket
+import com.keepit.search._
+import com.tzavellas.sse.guice.ScalaModule
+
+import play.api.Play.current
+
+class S3Module() extends ScalaModule with Logging {
```
Comment (eishay) [inline shoebox/app/com/keepit/module/S3Module.scala:19]: nice, much cleaner

**Comment by eishay** [comment]
like

**Comment by gmethvin** [comment]
closing for now as I refactor some things

**Pull request closed without merging**